### PR TITLE
Make Python 3.5 the default Python environment.

### PR DESCRIPTION
**Pull request by henri-hulski**
I'm going here consequently one step further then planned and proposing to make Python 3.5 the default Python environment.

### CODE REVIEW

**Comment by henri-hulski** [comment]
In core.py was used `# flake8: noqa`, which prevented the whole file to be tested by flake8.
I changed this to `# noqa` which ignores only the specific line.
As a result I get many unused imports errors:
```python
import dectate
import importscan
from webob import Response as BaseResponse, BaseRequest
from . import directive  # install directives with App
from .view import View
from .request import Request, Response
```
I made flake8 ignore the `dectate` and `directive` import as they seem to be necessary and removed the other ones. I also fixed some minor flake8 warnings.

@faassen  Please check if this is ok or if I have removed to much imports. The tests are all passing.

**Comment by henri-hulski** [comment]
I think, when we use the Python 3.5 syntax in the documentation examples it makes also sense to recommend to create a Python 3.5 environment for using and developing Morepath and also switch the pep8 and coverage tests to Python 3.5.

In more detail, if we add a note to `doc/developing.rst`, which says:

```
Throughout this documentation, examples are using the Python 3.5 syntax.
```
it's not very consistent if we instruct at the top of `doc/developing.rst` to:

```
First make sure you have virtualenv installed for Python 2.7.
```
and in `doc/installation.rst`:

```
To get started with Morepath right away, first create a Python 2.7 virtualenv.
```

So if we write our examples in Python 3.5 syntax we should also recommend to use a 3,5 environment and vice versa.

@faassen 
I think this changes are not very involved and it also don't breaks the Python 2.7 support, it only changes the recommendations. I think this PR contains all what is necessary and that's mainly adding or moving a few `# pragma: no cover` and `# noqa`.
So I think `dectate` and `reg` aren't involved at this point even if it makes sense to move there also.

**Comment by henri-hulski** [comment]
Summary of the changes this PR provides:

.  * Move the doc tests from Python 2.7 to 3.5, which involves:

    * Change the basepython for testenv:docs to Python 3.5 in tox.ini.
    * Add a note to `doc/developing.rst`. 
    * Fix `doc/testing.rst` and `reift.py` for Python 3.
    * Activate doc tests for JSON in settings.rst, as in Python 3 the response for YAML and JSON are consistent.
  * Add Python 3.5 testing environment to travis and tox. 
  * Add Python 3.5 classifier to setup.py. 
  * Change the recommendation for the virtual environment to Python 3.5.
  * Change basepython for flake8 test environment to 3.5, which involves adding 2 `# noqa`. 
  * Change the basepython for coverage to 3.5, which involves moving 2 and adding 2 `# pragma: no cover`.

**Comment by blaflamme** [comment]
@henri-hulski @faassen for python 3.5 I would opt for the [venv](https://docs.python.org/3/library/venv.html#module-venv) module instead of virtualenv

```
$ python3 -m venv myenv
```

**Comment by henri-hulski** [comment]
I think this is an interesting option as it's native and more lightweight then virtualenv. It seems also to have a simpler directory structure. 
But this is rather involved as it will be necessary to integrate with Travis and tox. 
So I think you should open a separate issue / PR for it.
Am 24.06.2016 17:59 schrieb Blaise Laflamme <notifications@github.com>:

File: morepath/reify.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def jammy(self):
       jammy called
       >>> print(v)
       1
-      >>> print f.jammy
+      >>> print (f.jammy)
```
Comment (taschini) [inline morepath/reify.py]: This is not [PEP8](https://www.python.org/dev/peps/pep-0008/) compliant:

```
Yes: spam(1)
No:  spam (1)
```

Comment (henri-hulski) [inline morepath/reify.py]: Sure. :-)  Thanks!

**Comment by henri-hulski** [comment]
Added PyPy 3 to tox test environments, as it's already used in dectate.

**Pull request merged**